jaxrs/src/main/java/com/cerner/beadledom/jaxrs/exceptionmappers/FailureExceptionMapper.java

```diff
@@ -0,0 +1,70 @@
+package com.cerner.beadledom.jaxrs.exceptionmappers;
+
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+
+import com.cerner.beadledom.jaxrs.models.JsonError;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.resteasy.spi.Failure;
+import org.jboss.resteasy.spi.ReaderException;
+import org.jboss.resteasy.spi.WriterException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Exception Mapper for all unhandled Exceptions extending from the {@link Failure} hierarchy
+ * (specifically for Resteasy Built-in Internally-Thrown Exceptions like {@link ReaderException},
+ * {@link WriterException}, etc.) thrown by the service. See
+ * http://docs.jboss.org/resteasy/docs/3.1.4.Final/userguide/html_single/index.html#builtinException
+ * for more information.
+ *
+ * @author Cal Fisher
+ */
+@Provider
+public class FailureExceptionMapper implements ExceptionMapper<Failure> {
+
+  private static final Logger logger = LoggerFactory.getLogger(FailureExceptionMapper.class);
+
+  /**
+   * Generates a Response object with a Status of the Failure, Content-Type 'application/json', and
+   * a JsonError entity containing details about the unhandled exception in Json format.
+   *
+   * @param exception the Failure exception that was not handled
+   * @return a json response with the exception's status or 500 if the exception has no response
+   */
+  @Override
+  public Response toResponse(Failure exception) {
```

Comment:
You aren't using the error code from the Failure. Do we want to propagate that into the json error?

Reply:
I think we do. I imagine something like

```java
@Provider
public class FailureExceptionMapper implements ExceptionMapper<Failure> {

  @Override
  public Response toResponse(Failure e) {
    Response.Status status = Response.Status.fromStatusCode(e.getErrorCode());

    if (status == null) {
      status = Response.Status.INTERNAL_SERVER_ERROR;
    }

    return Response
        .status(status)
        .type(MediaType.APPLICATION_JSON)
        .entity(
            JsonError.builder()
                .code(status.getStatusCode())
                .message(status.getReasonPhrase()))
        .build();
  }
}
```

but with the logging as well

Reply:
Used the error code from the Failure for the json error following a similar structure to @b-boogaard's suggestion above. Also included logging: https://github.com/cerner/beadledom/pull/45/commits/42cf0d727d134d6218f18ea1118901c89fa3a9d0#diff-94f0fc074c60a4a4d74f16bebf75afc9R44